langlib/langlib-test/src/test/resources/test-src/valuelib_test_negative.bal

```diff
@@ -30,3 +30,12 @@ function testCloneWithTypeWithInvalidInferredType() {
     MyClass|error y = x.cloneWithType();
     MyClass|error z = value:cloneWithType(x);
 }
+
+type Person record {
+    int name;
+};
+
+function testEnsureTypeFunction() returns error? {
+    Person p = {name: 12};
+    int age = check p.name;
```

Comment:
These tests should ideally go in a separate file, rather than in the langlib tests, since we're aren't directly testing the langlib. ensureType should have its own tests.

Reply:
We can do this separately.

Reply:
Created an issue to track this(https://github.com/ballerina-platform/ballerina-lang/issues/30009)